src/tool.rs

```diff
@@ -207,25 +214,29 @@ impl Tool {
                     &compiler_detect_output,
                 )?;
                 let stdout = String::from_utf8_lossy(&stdout);
-                guess_family_from_stdout(&stdout, path, cargo_output)
+                guess_family_from_stdout(&stdout, path, args, cargo_output)
             } else {
-                guess_family_from_stdout(&stdout, path, cargo_output)
+                guess_family_from_stdout(&stdout, path, args, cargo_output)
             }
         }
-        let detect_family = |path: &Path| -> Result<ToolFamily, Error> {
-            if let Some(family) = cached_compiler_family.read().unwrap().get(path) {
+        let detect_family = |path: &Path, args: &[String]| -> Result<ToolFamily, Error> {
+            let cache_key = (Box::from(path), args.to_vec());
+            if let Some(family) = cached_compiler_family.read().unwrap().get(&cache_key) {
                 return Ok(*family);
             }
 
-            let family = detect_family_inner(path, cargo_output, out_dir)?;
+            let family = {
+                let (path, args) = &cache_key;
+                detect_family_inner(path, args, cargo_output, out_dir)?
+            };
```

Comment:
I think we already have `path` and `args` as arguments, and since they are reference they should be still useable?

```suggestion
            let family = detect_family_inner(path, args, cargo_output, out_dir)?;
```

Reply:
Fixed as suggested with the latest force push; interdiff link: https://github.com/rust-lang/cc-rs/compare/c5c1a91ad6532e70081ead71ffc81485df116932..ffe0e4164b4d6ac31c62aed7888651d3d15f0c4f